apps/studio/components/interfaces/DataWarehouse/CreateWarehouseCollection.tsx

```diff
@@ -0,0 +1,98 @@
+import { useParams } from 'common'
+import { useCreateCollection } from 'data/analytics/warehouse-collections-create-mutation'
+import { useRouter } from 'next/router'
+import React from 'react'
+import toast from 'react-hot-toast'
+import { Button, IconPlus, Input, Modal } from 'ui'
+
+type Props = {}
+
+export const CreateWarehouseCollectionModal = (props: Props) => {
+  const [isOpen, setIsOpen] = React.useState(false)
+  const router = useRouter()
+  const { ref } = useParams()
+
+  const {
+    mutateAsync: createCollection,
+    isLoading,
+    data: newCollection,
+  } = useCreateCollection({
+    projectRef: ref || 'default',
```

Comment:
good point - not yet and we should consider this so its not repeated everywhere

on self-hosted by default the ref should already be `default` since the route for self-hosted is /default/..., so technically || 'default' is unnecessary but could we double check this still?